### PR TITLE
 Add AsyncReadTestExt::limited 

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**
Also merges `InterleavePending` and `InterleavePendingWrite` to reduce code duplication. And renamed `LimitedWrite` to `Limited` and implemented `AsyncRead` on it for `AsyncReadTestExt::limited `.

### CODE REVIEW

File: futures-test/src/io/interleave_pending.rs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+};
+
+/// I/O wrapper for interleaving [`Poll::Pending`] in calls to read or write.
+///
+/// See the [`interleave_pending`] and [`interleave_pending_write`] methods.
+///
+/// [`interleave_pending`]: super::AsyncReadTestExt::interleave_pending
+/// [`interleave_pending_write`]: super::AsyncWriteTestExt::interleave_pending_write
+#[derive(Debug)]
+pub struct InterleavePending<IO> {
```
Comment (Nemo157) [inline futures-test/src/io/interleave_pending.rs]: I'm not sure what the standard is for 2 character acronyms in Rust's version of PascalCase/UpperCamelCase. As far as I know there's no special-case for them so they should follow the standard acronym rule of just treating it as a single word, which would make this `Io`. (If I remember my .NET conventions correctly they had a special case for < 3 character acronyms to be kept uppercase like this).

Comment (Thomasdezeeuw) [reply futures-test/src/io/interleave_pending.rs]: Done.

File: futures-test/src/io/interleave_pending.rs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+};
+
+/// I/O wrapper for interleaving [`Poll::Pending`] in calls to read or write.
+///
+/// See the [`interleave_pending`] and [`interleave_pending_write`] methods.
+///
+/// [`interleave_pending`]: super::AsyncReadTestExt::interleave_pending
+/// [`interleave_pending_write`]: super::AsyncWriteTestExt::interleave_pending_write
+#[derive(Debug)]
+pub struct InterleavePending<Io> {
```
Comment (cramertj) [inline futures-test/src/io/interleave_pending.rs]: Maybe a silly question: can we also use this for `Future`s / `Stream`s / `Sink`s?

Comment (Thomasdezeeuw) [reply futures-test/src/io/interleave_pending.rs]: Sure, want me to that in this pr?

Comment (cramertj) [reply futures-test/src/io/interleave_pending.rs]: That would be great, thanks! :)

**Comment by cramertj** [comment]
Great, thanks!

**Pull request merged**